pkg/certs/util.go

```diff
@@ -68,3 +71,58 @@ func CertificateBundleToString(bundle []*x509.Certificate) string {
 	}
 	return strings.Join(output, "\n")
 }
+
+func ValidatePrivateKey(pemKey []byte) []error {
+	if len(pemKey) == 0 {
+		return []error{fmt.Errorf("required private key is empty")}
+	}
+	if _, err := keyutil.ParsePrivateKeyPEM(pemKey); err != nil {
+		return []error{fmt.Errorf("failed to parse the private key, %w", err)}
+	}
+	return []error{}
```

Comment:
nit: for consistency let's `return nil`.

Reply:
Actually, that `nil` from `ValidateServerCert` is inconsistent. Let me fix that one instead of this.

Reply:
If you want to return empty slices, returning nil is more idiomatic. You avoid memory allocation.

Reply:
Done